cmake/onnxruntime_python.cmake

```diff
@@ -76,15 +58,7 @@ else()
   set(ONNXRUNTIME_SO_LINK_FLAG "-DEF:${ONNXRUNTIME_ROOT}/python/pybind.def")
 endif()
 
-if (onnxruntime_ENABLE_TRAINING)
-  target_link_libraries(onnxruntime_pybind11_state PRIVATE onnxruntime_training)
-endif()
-
-if (onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
-  target_link_libraries(onnxruntime_pybind11_state PRIVATE onnxruntime_interop_torch)
-endif()
-
-target_link_libraries(onnxruntime_pybind11_state PRIVATE
```

Comment:
This part should be similar to the things in cmake/onnxruntime.cmake. If you make any changes here, usually you should change cmake/onnxruntime.cmake too. 

You are not only add new functionalities. You are changing the dependency orders of the existing inference-only libraries.

Reply:
I made the change, and I reverted as we talked. because onnxruntime.cmake is not needed for training at this phase.

Reply:
So what about I add Training flag for those re-ordering?

Reply:
this is the error when I did not re-order onnxruntime_optimizer.cmake, it is okay to build, but seems python import failed. 
```

ImportError while importing test module '/home/pengwa/dev/onnxruntime/orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/conda/lib/python3.7/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
orttraining/orttraining/test/python/orttraining_test_ortmodule_autograd.py:5: in <module>
    import onnxruntime
/opt/conda/lib/python3.7/site-packages/onnxruntime/__init__.py:34: in <module>
    raise import_capi_exception
/opt/conda/lib/python3.7/site-packages/onnxruntime/__init__.py:23: in <module>
    from onnxruntime.capi._pybind_state import get_all_providers, get_available_providers, get_device, set_seed, \
/opt/conda/lib/python3.7/site-packages/onnxruntime/capi/_pybind_state.py:67: in <module>
    from .onnxruntime_pybind11_state import *  # noqa
E   ImportError: /opt/conda/lib/python3.7/site-packages/onnxruntime/capi/onnxruntime_pybind11_state.so: undefined symbol: _ZTVN11onnxruntime19InsertMaxPoolOutputE
============================================================================================ short test summary info ================================================
```

Reply:
Oh i found how to resolve this without re-ordering.

Reply:
Then where does "onnxruntime::InsertMaxPoolOutput" from and who was using it?

Reply:
onnxruntime::InsertMaxPoolOutput is implemented in onnxruntime_optimizers and it is used by orttraining/orttraining/core/framework/gradient_graph_builder.cc which is part of onnxruntime_training. onnxruntime_training is on top of all the other inference libs. So if the original setting works fine.